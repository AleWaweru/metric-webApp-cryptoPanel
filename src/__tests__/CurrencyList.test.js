import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import CurrencyList from '../Components/Main/CurrencyList';

describe('CurrencyList', () => {
  const currencies = [
    // currency data
  ];

  const query = '';
  const handleQuery = jest.fn();
  const handleKeyDown = jest.fn();

  it('renders the component correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <CurrencyList
          currencies={currencies}
          query={query}
          handleQuery={handleQuery}
          handleKeyDown={handleKeyDown}
        />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
