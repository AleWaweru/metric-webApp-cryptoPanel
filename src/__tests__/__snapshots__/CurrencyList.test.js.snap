// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyList renders the component correctly 1`] = `
<div>
  <div
    class="search-bar"
  >
    <input
      placeholder="Search"
      type="text"
      value=""
    />
    <button
      type="button"
    >
      Search
    </button>
  </div>
  <div
    class="table-container"
  >
    <table>
      <thead>
        <tr>
          <th>
            Rank
          </th>
          <th>
            Name
          </th>
          <th>
            Symbol
          </th>
          <th>
            Current Price
          </th>
          <th>
            24h Change
          </th>
          <th>
            Total Volume
          </th>
          <th>
            Market Cap
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
